addons/website_slides/static/src/js/public/components/slide_upload_dialog/slide_upload_dialog.js

```diff
@@ -0,0 +1,631 @@
+                    this.file.type === "image/svg+xml" ? await this._svgToPNG() : this.file.data,
+            });
+        }
+        return values;
+    }
+
+    /**
+     * @private
+     */
+    _onFormSubmitDone(data, oldType) {
+        if (data.error) {
+            this.state.page = oldType;
+            this.state.size = "lg";
+            this._alertDisplay(data.error);
+        } else {
+            if (data.url.indexOf("enable_editor") >= 0) {
+                // If we need to enter edit mode, it should be done to the top
+                // window so that we endup refreshing the backend client action
+                // in edit mode.
+                const { origin, pathname } = window.top.location;
+                const _url = new URL(data.url, `${origin}${pathname}`);
+                if (_url.origin === origin) {
+                    window.top.location = encodeURI(_url.href);
+                }
+            } else {
+                redirect(data.url);
+            }
+        }
+    }
+
+    /**
+     * Get value for category_id and tag_ids (ORM cmd) to send to server
+     *
+     * @private
+     */
+    _getDropdownValues() {
+        const result = {};
+        // tags
+        if (this.choices.tagIds.length > 0) {
+            result["tag_ids"] = this.choices.tagIds.map((tagId) => {
+                const tag = this.choices.tags.find((tag) => tag.value === tagId);
+
+                if (this._toCreate(tag.value)) {
+                    return [0, 0, { name: tag.label }];
+                } else {
+                    return [4, tag.value];
+                }
+            });
+        }
+        // category
+        if (!this.defaultCategoryID) {
+            if (this._toCreate(this.choices.categoryId)) {
+                const category = this.choices.categories.find((cat) => {
+                    return cat.value === this.choices.categoryId;
+                });
+                result["category_id"] = [0, { name: category.label }];
+            } else {
+                const categoryId = parseInt(this.choices.categoryId, 10);
+                result["category_id"] = [categoryId];
+                this.categoryID = categoryId;
+            }
+        } else {
+            result["category_id"] = [this.defaultCategoryID];
+        }
+        return result;
+    }
+
+    /**
+     * Get a temporary id for new category or tag
+     * @private
+     */
+    _getTempId() {
+        return `temp${Math.floor(Math.random() * 999)}`;
+    }
+
+    /**
+     * @private
+     */
+    _toCreate(value) {
+        return typeof value === "string" && value.startsWith("temp");
+    }
+
+    /**
+     * @private
+     */
+    async _fetch_choices(type, domain = [], fields = ["name"]) {
+        const choices = [];
+        const results = await this.rpc(`/slides/${type}/search_read`, { fields, domain });
+
+        results.read_results.forEach((choice) => {
+            choices.push({ value: choice.id, label: choice.name });
+        });
```

Comment:
You could just return `results.read_results` to avoid computation

Reply:
I'm not sure how to avoid remapping `read_results`. How should I make this results match the `SelectMenu` choices/choice props?